main.groovy

```diff
@@ -0,0 +1,349 @@
+                            name: 'jdk_list',
+                            defaultValue: '',
+                            description: 'The JDK and its version. If there are multiple parameters, please add them by separating them by a ","(Comma). Check the README file on https://github.com/wso2/testgrid-jenkins-library/tree/main/README.md',
+                            trim: false
+                        ),
+                        string(
+                            name: 'database_list',
+                            defaultValue: '',
+                            description: 'The Database type and its version. If there are multiple parameters, please add them by separating them by a ","(Comma). Check the README file on https://github.com/wso2/testgrid-jenkins-library/tree/main/README.md',
+                            trim: false
+                        ),
+                        booleanParam(
+                            name: 'use_wum',
+                            defaultValue: false,
+                            description: 'If using WUM this should be true. If using U2 this should be false. Check the README file on https://github.com/wso2/testgrid-jenkins-library/tree/main/README.md'
+                        ),
+                        string(
+                            name: 'product_repository',
+                            defaultValue: '',
+                            description: 'The product repo where the test scripts are existing. Check the README file on https://github.com/wso2/testgrid-jenkins-library/tree/main/README.md',
+                            trim: false
+                        ),
+                        string(
+                            name: 'product_test_branch', 
+                            defaultValue: '',
+                            description: 'The repo branch where the test script is existing. Check the README file on https://github.com/wso2/testgrid-jenkins-library/tree/main/README.md',
+                            trim: false
+                        ),
+                        string(
+                            name: 'product_test_script', 
+                            defaultValue: '',
+                            description: 'The location of the test script. Check the README file on https://github.com/wso2/testgrid-jenkins-library/tree/main/README.md',
+                            trim: false
+                        )
+                    ])
+                ])
+                aws_repo_branch="" //need to exit if the branch name is invalid
+                if (use_wum.toBoolean()){
+                    aws_repo_branch="${product_version}-new"
+                    updateType="wum"
+                }else{
+                    aws_repo_branch="${product_version}-u2-new"
+                    updateType="u2"
+                }
+                dir("aws-"+product) {
+                    git branch: "${aws_repo_branch}",
+                    credentialsId: "JANETH_GITHUB_TEMP",
```

Comment:
Shall we configure new Credentials(wso2 common user) for a GitHub user and use the ID here.

Reply:
Fixed by replacing to the new id `WSO2_GITHUB_TOKEN`